swap-logic/src/swap_control/swap_storage.rs

```diff
@@ -0,0 +1,201 @@
+use crate::constants::{keys, users};
+use alloc::{
+    collections::BTreeMap,
+    string::{String, ToString},
+};
+use contract::{
+    contract_api::{runtime, storage as contract_storage},
```

Comment:
Seems okay to use the contract_storage as storage.

Reply:
https://github.com/hdac-io/swap/pull/1/commits/9043590ed4061a0b04b323918365edc58b9e1af4